lib/vdc/DCBOV.cpp

```diff
@@ -0,0 +1,317 @@
+#include <vector>
+#include <algorithm>
+#include <map>
+#include <iostream>
+#include <fstream>
+#include "vapor/VAssert.h"
+#include <stdio.h>
+
+#ifdef _WINDOWS
+    #define _USE_MATH_DEFINES
+    #pragma warning(disable : 4251 4100)
+#endif
+#include <cmath>
+
+#include <vapor/GeoUtil.h>
+#include <vapor/UDUnitsClass.h>
+#include <vapor/DCBOV.h>
+#include <vapor/DCUtils.h>
+
+using namespace VAPoR;
+using namespace std;
+
+DCBOV::DCBOV() : _bovCollection(nullptr)
+{
+    _dimsMap.clear();
+    _coordVarsMap.clear();
+    _dataVarsMap.clear();
+    _meshMap.clear();
+    _coordVarKeys.clear();
+}
+
+DCBOV::~DCBOV()
+{
+    if (_bovCollection != nullptr) { delete _bovCollection; }
+}
+
+int DCBOV::initialize(const vector<string> &paths, const std::vector<string> &options)
+{
+    _bovCollection = new BOVCollection();
+    int rc = _bovCollection->Initialize(paths);
+    if (rc < 0) {
+        SetErrMsg("Failure reading .bov file");
+        return (-1);
+    }
+
+    //
+    //  Get the dimensions of the grid.
+    //	Initializes members: _dimsMap
+    //
+    _InitDimensions();
+
+    // Set up the coordinate variables
+    //
+    _InitCoordinates();
+
+    // Identify data and coordinate variables. Sets up members:
+    // Initializes members: _dataVarsMap, _meshMap
+    //
+    _InitVars();
+
+    return (0);
+}
+
+void DCBOV::_InitDimensions()
+{
+    _dimsMap.clear();
+    std::array<std::string, 3> dimnames = _bovCollection->GetSpatialDimensions();
+    std::array<size_t, 3>      dimlens = _bovCollection->GetDataSize();
+
+    for (int i = 0; i < dimnames.size(); i++) {
+        Dimension dim(dimnames[i], dimlens[i]);
+        _dimsMap[dimnames[i]] = dim;
+    }
+
+    string    timeDim = _bovCollection->GetTimeDimension();
+    size_t    numTimes = _bovCollection->GetUserTimes().size();
+    Dimension dim(timeDim, numTimes);
+    _dimsMap[timeDim] = dim;
+}
+
+void DCBOV::_InitCoordinates()
+{
+    bool                       uniformHint = true;
+    vector<bool>               periodic(false, 4);
```

Comment:
Can you 1) state the intend of this vector initialization, and 2) verify that the argument ordering is correct?

Reply:
1) to configure a non-periodic coordinate variable, 2) It's not, 3) Not sure.

Thanks for the catch.  Fixed.

Reply:
If the intent is to create a vector of size 4, and in reality a vector of size 0 is created which is the case here, and the rest of the program behaves as if a vector of size 4 is created, then that almost certainly suggests that another bug exists somewhere. I encourage you to investigate thoroughly and rule out other potential bugs.